modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointDirtyPages.java

```diff
@@ -167,15 +180,15 @@ public IgniteConcurrentMultiPairQueue<PersistentPageMemory, GroupPartitionId> to
         FullPageId[] pageIds = dirtyPages.get(regionIndex).dirtyPages.pageIds;
 
         if (fromPosition == pageIds.length - 1 || !equalsByGroupAndPartition(pageIds[fromPosition], pageIds[fromPosition + 1])) {
```

Comment:
This is weirdly specific case, why did you add this code? Does it break with regular binary searches?
Again, this makes code more complicated. That's why I'm asking. If we can just remove it then let's remove it

Reply:
In general, this is a small optimization, but if you insist, I will remove it.

Reply:
I don't think that it'll have any noticeable effect, while making code more complicated. Yes, I insist

Reply:
Okay, we're afraid of more complex code, removed that.